### PR TITLE
[AIRFLOW-2462] Change PasswordUser pasword setter to correct syntax

### PR DESCRIPTION
**Pull request by Noremac201**
Make sure you have checked _all_ steps below.

### JIRA
- [x] My PR addresses the following [Airflow JIRA](https://issues.apache.org/jira/browse/AIRFLOW/) issues and references them in the PR title. For example, "\[AIRFLOW-XXX\] My Airflow PR"
    - https://issues.apache.org/jira/browse/AIRFLOW-2462

### Description
- [x] Here are some details about my PR, including screenshots of any UI changes:
Fixed incorrect sqlalchemy setter syntax in the PasswordUser password setter

### Tests
- [x] My PR adds the following unit tests __OR__ does not need testing for this extremely good reason:

### Commits
- [x] My commits all reference JIRA issues in their subject lines, and I have squashed multiple commits if they address the same issue. In addition, my commits follow the guidelines from "[How to write a good git commit message](http://chris.beams.io/posts/git-commit/)":
    1. Subject is separated from body by a blank line
    2. Subject is limited to 50 characters
    3. Subject does not end with a period
    4. Subject uses the imperative mood ("add", not "adding")
    5. Body wraps at 72 characters
    6. Body explains "what" and "why", not "how"

### Documentation
- [x] In case of new functionality, my PR adds documentation that describes how to use it.
    - When adding new operators/hooks/sensors, the autoclass documentation generation needs to be added.

### Code Quality
- [x] Passes `git diff upstream/master -u -- "*.py" | flake8 --diff`

### CODE REVIEW

**Comment by ashb** [comment]
Lets have some tests of this so we notice if it breaks in the future please!

**Comment by Noremac201** [comment]
Added tests; the build shows failing, my local Travis-CI shows green.
https://travis-ci.org/Noremac201/incubator-airflow/builds/383850999

File: tests/core.py [hunk 1]
```diff
@@ -1865,6 +1865,31 @@ def tearDown(self):
...
+
+    def test_password_unicode(self):
+        # In python2.7 no conversion is required back to str
+        # In python >= 3 the method must convert from bytes to str
+        self.password_user.password = "secure_password"
+        self.assertIsInstance(self.password_user.password, str)
+
+    def test_password_user_authenticate(self):
+        self.password_user.password = "secure_password"
+        self.assertTrue(self.password_user.authenticate("secure_password"))
```
Comment (ashb) [inline tests/core.py:1890]: Could you add one more test - like this one, but that round-trips through the database.

Comment (Noremac201) [reply tests/core.py:1890]: Sure, just a little clarification, is it any different than these tests?
https://github.com/apache/incubator-airflow/blob/master/tests/core.py#L1868-L1932

Comment (ashb) [reply tests/core.py:1890]: Sort of? Those tests are testing the auth of the webserver. I am happy with a separate test of the PasswordUser model (as you have here) or of them being merged into the same TestCase.

Though I am somewhat wondering why those tests weren't failing on master.... curious.

**Comment by Noremac201** [comment]
Travis being finicky again, my builds showing green:
https://travis-ci.org/Noremac201/incubator-airflow/builds/385834811

**Review by ashb — APPROVED** [review]

**Review by Fokko — APPROVED** [review]

**Pull request closed without merging**